homeassistant/components/apsystems/entity.py

```diff
@@ -0,0 +1,33 @@
+"""APsystems base entity."""
+
+from __future__ import annotations
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ApsystemsConfigEntry, ApSystemsData
+from .const import DOMAIN
+from .coordinator import ApSystemsDataCoordinator
+
+
+class ApsystemsEntity(CoordinatorEntity[ApSystemsDataCoordinator]):
+    """Defines a base APsystems entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        data: ApSystemsData,
+        entry: ApsystemsConfigEntry,
+    ) -> None:
+        """Initialize the APsystems entity."""
+        super().__init__(data.coordinator)
+        self._entry = entry
+        assert entry.unique_id
+        self._attr_unique_id = entry.unique_id
+        self._attr_device_info = DeviceInfo(
```

Comment:
I think it would be nice if the base entity has the same structure as the current sensor and that we approach the api via the coordinator in the new platforms.

Reply:
So you say that the coordinator should store all the data? I disagree. I am planning to add 2 new platforms, each requiring a separate endpoint which tends to throw a Timeout from time to time. So IMO it doesn't make sense to "sync" the updates when each request takes from 1-6 seconds.

Reply:
Right, it was just a proposal, but since we don't need the API here yet, I think we can keep the entry runtime data the same and just keep it minimal for now so then we can dive in the new context in the followup PR

Reply:
I personally would just leave it like that as I am planning to propose the new PRs tomorrow or in the next days, as I'd just roll back the changes to the first commit of this PR when adding the number.

Reply:
Continue building that PR on this PR, and open a PR for it. I think I am missing something or you're seeing something I don't see. So I don't want to approve this one just yet because I don't want to refactor that one again if it turns out to that it should be changed